framework/aster-frame/src/vm/config.rs

```diff
@@ -0,0 +1,40 @@
+// SPDX-License-Identifier: MPL-2.0
+
+//! A configuration module for framework memory management.
+
+use super::Vaddr;
+
+pub const KERNEL_STACK_SIZE: usize = PAGE_SIZE * 64;
+pub const KERNEL_HEAP_SIZE: usize = PAGE_SIZE * 256;
+
+/// Typicall 64-bit systems have at least 48-bit virtual address space.
+/// Halving it to reserve half of the address space for the kernel.
+/// 
+/// Also, the top page is not regarded as usable since it's a workaround
+/// for the x86_64 CPUs' bugs. See
+/// <https://github.com/torvalds/linux/blob/480e035fc4c714fb5536e64ab9db04fedc89e910/arch/x86/include/asm/page_64.h#L68-L78>
+/// for the rationale.
+pub const USERSPACE_LOWEST_UNUSABLE_VADDR: Vaddr = 0x0000_8000_0000_0000 - PAGE_SIZE;
+
+// FIXME: the following constants are all architecture dependent. Need
+// a proper redesign.
```

Comment:
So why not move them to `arch/x86/mm/mod.rs` or `arch/x86/mm/params.rs`?

The const values defined here is really arch dependent.

And yes, I do not think `config.rs` is the right file name. A configuration is more like a tunable value. A configuration lets you to make certain tradeoffs. But you cannot tune the page size of x86. And changing the base address of the direct mapping of physical memory does not provide you performance advantage. So these values are not configurations. "Parameters" is probably the better word.

Last, please hide these values from the users of Framework if the users do not need to care about them. I really don't think all of these const values should be public items in a public module.